### DISCUSSION TITLE
Public API Documentation?

### DISCUSSION THREAD

**Discussion by n-sweep** [johnwmillr/LyricsGenius#234] [Q&A] [answered]
I've been implementing a lyrics-searching tool based on the Genius API documentation at `https://docs.genius.com` and ended up here after becoming frustrated at the seemingly limited documentation/capability of the API. The docs seem to imply that `https://api.genius.com` is the only root for API access, and I only found the public `https://genius.com/api` root after digging in to this (wonderful) library.

Does documentation exist somewhere for the public API at `genius.com/api`? It seems much more robust, with the endpoints like `/search/artists` and `/search/albums` rather than the very limited songs-only search documented at genius.

Thank you!

**Comment by allerter** [comment, marked as answer]
Hi, sorry for the late answer. The public API -or Web API as Genius calls it- is meant for users. That's why it has so many capabilities, and also why there is no documentation available for it because it's not meant to be used by developers. The documentation and the endpoints that we've added here in LyricsGenius, we added through trial-and-error. This is what one of Genius' staff had to say about [using the Web API](https://genius.com/discussions/267781-Search-api-results):
> Using the web API for non-browser based requests is not supported, and people found to be misusing it to gain information not provided by our public API may be banned at any time with no warning.

**Reply by n-sweep** [reply]
Thanks very much for the info and the context, it's much appreciated.
